src/App.jsx

```diff
@@ -1,19 +1,118 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
+import { useEffect, useState } from 'react'
+
 import './App.css'
 import TodoList from './TodoList'
 import AddTodoForm from '../AddTodoForm'
 
 function App() {
-  const [newTodo, setNewTodo] = useState('')
+  //  const useSemiPersistentState = () => {
+  let [todoList, setTodoList] = useState([])
+  let [isLoading, setIsLoading] = useState(true)
+
+  async function fetchData() {
+    const options = {
+      method: 'GET',
+      headers: {
+        Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`,
+      },
+    }
+    const url = `https://api.airtable.com/v0/${
+      import.meta.env.VITE_AIRTABLE_BASE_ID
+    }/${import.meta.env.VITE_TABLE_NAME}`
+
+    try {
+      const response = await fetch(url, options)
+
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`)
+      }
+
+      const data = await response.json()
+
+      const todos = data.records.map((record) => ({
+        id: record.id,
+        title: record.fields.title,
+      }))
+
+      setTodoList(todos)
+      setIsLoading(false)
+    } catch (error) {
+      console.error(error.message)
+    }
+  }
+
+  useEffect(() => {
+    fetchData()
+  }, [])
+
+  useEffect(() => {
+    const loadTodoList = () => {
+      return new Promise((resolve) => {
+        setTimeout(() => {
+          const savedTodoList = localStorage.getItem('savedTodoList')
+          const initialList = savedTodoList ? JSON.parse(savedTodoList) : []
+          resolve({ data: { todoList: initialList } })
+        }, 2000)
+      })
+    }
+    loadTodoList().then((result) => {
+      setTodoList(result.data.todoList)
+      setIsLoading(false)
+    })
+  }, [])
+
+  useEffect(() => {
+    if (!isLoading) {
+      localStorage.setItem('savedTodoList', JSON.stringify(todoList))
+      console.log(todoList)
+    }
+  }, [todoList, isLoading])
+
+  // useEffect(() => {
```

Comment:
Make sure to remove commented-out code from your app when you no longer need it. If you ever need to come back and refer to it, you can always find it in your commit history – so you can safely delete it without losing it permanently.